test/libdnf/rpm/test_package.cpp

```diff
@@ -0,0 +1,346 @@
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(conflicts));
+}
+
+
+void RpmPackageTest::test_get_obsoletes() {
+    const auto obsoletes = get_pkg("npm-1:5.12.1-1.fc29.x86_64").get_obsoletes();
+    const std::vector<std::string> expected = {"npm < 3.5.4-6"};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(obsoletes));
+}
+
+
+void RpmPackageTest::test_get_prereq_ignoreinst() {
+    const auto ipr = get_pkg("npm-1:5.12.1-1.fc29.x86_64").get_prereq_ignoreinst();
+    // TODO no prereq_ignoreinst (called Ipr: in libsolv testcases) in test packages
+    const std::vector<std::string> expected = {};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(ipr));
+}
+
+
+void RpmPackageTest::test_get_regular_requires() {
+    const auto reg_reqs = get_pkg("filesystem-3.9-2.fc29.x86_64").get_regular_requires();
+    // TODO doesn't test a difference between requires() and this, since there are no prereqs
+    const std::vector<std::string> expected = {"setup"};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(reg_reqs));
+}
+
+
+void RpmPackageTest::test_get_recommends() {
+    const auto recommends = get_pkg("abcde-2.9.2-1.fc29.noarch").get_recommends();
+    const std::vector<std::string> expected = {"flac"};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(recommends));
+}
+
+
+void RpmPackageTest::test_get_suggests() {
+    const auto suggests = get_pkg("abcde-2.9.2-1.fc29.noarch").get_suggests();
+    const std::vector<std::string> expected = {"lame"};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(suggests));
+}
+
+
+void RpmPackageTest::test_get_enhances() {
+    const auto enhances = get_pkg("abcde-2.9.2-1.fc29.noarch").get_enhances();
+    // TODO no enhances in test packages
+    const std::vector<std::string> expected = {};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(enhances));
+}
+
+
+void RpmPackageTest::test_get_supplements() {
+    const auto supplements = get_pkg("glibc-langpack-af-2.28-9.fc29.x86_64").get_supplements();
+    const std::vector<std::string> expected = {"(glibc and (langpacks-af or langpacks-af_ZA))"};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(supplements));
+}
+
+
+void RpmPackageTest::test_get_baseurl() {
+    CPPUNIT_ASSERT_EQUAL(std::string(), get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_baseurl());
+}
+
+
+void RpmPackageTest::test_get_location() {
+    CPPUNIT_ASSERT_EQUAL(
+        std::string("x86_64/CQRlib-1.1.1-4.fc29.x86_64.rpm"),
+        get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_location()
+    );
+}
+
+
+void RpmPackageTest::test_get_checksum() {
+    CPPUNIT_ASSERT_EQUAL(
+        std::string("8b1968310a8409bbca2c06d0217b010e83a82fbb515627d87c52ef7ba084e8b3"),
+        get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_checksum().get_checksum()
+    );
+}
+
+
+void RpmPackageTest::test_get_hdr_checksum() {
+    // TODO is it correct it's empty?
+    CPPUNIT_ASSERT_EQUAL(std::string(""), get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_hdr_checksum().get_checksum());
```

Comment:
HEADER checksum is available only for installed packages, therefore it is correct.

Reply:
Please the note must be in format TODO, then add tests for installed packages

Reply:
Again, not sure what you mean. I've added "TODO" to the comment (the snippet github now shows is outdated too).